src/components/widget/widget.component.ts

```diff
@@ -118,7 +120,11 @@ const WidgetComponent: ng.IComponentOptions = {
         .apply(this.widget.chart.service.caller, args)
         .then(response => {
           this.results = response.data;
-          if (this.widget.chart.percent) {
+          if (this.widget.chart.type === 'line' && this.$state.params.apiId) {
+            this.EventsService.search(['PUBLISH_API'], [this.$state.params.apiId], response.data.timestamp.from, response.data.timestamp.to, 0, 10).then(response => {
+              this.results.events = response.data;
+            }).finally(() => this.fetchData = false);
```

Comment:
This can be mutualized with the other of `this.fetchData = false` the if/else clauses:
 - Move the finally to L140
 - Remove the `else` part L137
 - add `return` before the service call on L124 and L130

Let me know if it's unclear or if you want to pair on that

Reply:
good catch !

Reply:
resolved